### PR TITLE
Fix argument check for the generator command.

### LINKED ISSUE #5412: Logstash Plugin Generator Allows creationg of plugin without setting --type
new logstash plugin generator errors out when creating new plugin

```
~/Code/es/talevy-logstash (master)$ bin/logstash-plugin generate --name codec-confluent-avro
 Creating /Users/tal/Code/es/talevy-logstash/logstash--codec-confluent-avro
ERROR: Plugin creation Aborted, message: No such file or directory - No such directory: /Users/tal/Code/es/talevy-logstash/lib/pluginmanager/templates/-plugin
```

Expected:

- no error
- and creating `logstash-codec-confluent-avro` instead of `logstash--codec-confluent-avro`

### PR DESCRIPTION
**Pull request by purbon**
* add more control on valid arguments for codecs in the generator command.

Partially fixed #5412 

follow up PR adding codecs is on its way.

### CODE REVIEW

**Comment by suyograo** [comment]
Lets wait for codec support so we can fix this right once for all.

**Comment by purbon** [comment]
both fix are independent, you still need to validate the right values for this field.

**Comment by suyograo** [comment]
Yeah, understood, but why not add codec support first and then fix https://github.com/elastic/logstash/pull/5419/files#diff-1d7289384dcead3ba670b14383780487R13
to include codec?

**Comment by purbon** [comment]
check #5421, it might take long to actually decide if the codec example is good enough, so this is why I break up the fix, one for fixing each problem.

File: lib/pluginmanager/generate.rb [hunk 1]
```diff
@@ -35,6 +32,10 @@ def execute
 
   private
 
+  def validate_params
+    raise(ArgumentError, "should be one of: input, output or filter") unless TYPES.include?(type)
```
Comment (talevy) [inline lib/pluginmanager/generate.rb]: will be `input, output, filter, or codec` once the other PR is in. yeah?

Comment (untergeek) [inline lib/pluginmanager/generate.rb]: Yep.  See #5421

**Comment by untergeek** [comment]
LGTM™

File: lib/pluginmanager/generate.rb (hunk 1 above)
Comment (purbon) [inline lib/pluginmanager/generate.rb]: agreed, just tried to split the PR as I was not expecting the codec to go in so fast 👍

**Pull request closed without merging**